client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -238,7 +238,8 @@ const Checkout = React.createClass( {
 			return selectedSiteSlug
 				? `/plans/${ selectedSiteSlug }/thank-you`
 				: '/checkout/thank-you/plans';
-		} else if ( selectedSite.options.is_domain_only && cartItems.hasDomainRegistration( cart ) && ! cartItems.hasPlan( cart ) ) {
+		} else if ( selectedSite.options && selectedSite.options.is_domain_only &&
```

Comment:
You could use the [`getSiteOption` selector](https://github.com/Automattic/wp-calypso/blob/cac870ea9f22176ae0f3bdc9371fddc598532ea2/client/state/sites/selectors.js#L309-L324) here.